pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/LawOfDemeterRule.java

```diff
@@ -33,22 +32,20 @@
 import net.sourceforge.pmd.lang.symboltable.NameDeclaration;
 import net.sourceforge.pmd.lang.symboltable.Scope;
 
-/**
- * This rule can detect possible violations of the Law of Demeter. The Law of
- * Demeter is a simple rule, that says "only talk to friends". It helps to
- * reduce coupling between classes or objects.
- * <p>
- * See:
- * <ul>
- * <li>Andrew Hunt, David Thomas, and Ward Cunningham. The Pragmatic Programmer.
- * From Journeyman to Master. Addison-Wesley Longman, Amsterdam, October
- * 1999.</li>
- * <li>K.J. Lieberherr and I.M. Holland. Assuring good style for object-oriented
- * programs. Software, IEEE, 6(5):38–48, 1989.</li>
- * </ul>
```

Comment:
Please restore the complete class comment.

What you have written as a class comment should actually go into your commit message.... (you are explaining what you have done and why....).

Also, there is no javadoc tag `@version`. Please keep `@since` - it says, that the LawOfDemeter rule exists since PMD 5.0. Also, a class cannot `@return` anything....